packages/jest-docblock/src/index.js

```diff
@@ -35,17 +47,64 @@ function parse(docblock: string): {[key: string]: string} {
   let prev = '';
   while (prev !== docblock) {
     prev = docblock;
-    docblock = docblock.replace(multilineRe, '\n$1 $2\n');
+    docblock = docblock.replace(multilineRe, `${line}$1 $2${line}`);
   }
   docblock = docblock.trim();
 
   const result = Object.create(null);
+  const comments = docblock.replace(propertyRe, '').replace(lineStartRe, line);
   let match;
   while ((match = propertyRe.exec(docblock))) {
     result[match[1]] = match[2];
   }
-  return result;
+  return {comments: comments.trim(), pragmas: result};
+}
+
+export function print({
+  comments = '',
+  pragmas = {},
+}: {
+  comments?: string,
+  pragmas?: {[key: string]: string},
+}): string {
+  const line = detectNewline(comments) || EOL;
+  const head = '/**';
+  const start = ' *';
+  const tail = ' */';
+
+  const keys = Object.keys(pragmas);
+
+  const printedObject = keys
+    .map(key => start + ' ' + printKeyValue(key, pragmas[key]) + line)
+    .join('');
+
+  if (!comments) {
+    if (keys.length === 0) {
+      return '';
+    }
+    if (keys.length === 1) {
+      return `${head} ${printKeyValue(keys[0], pragmas[keys[0]])}${tail}`;
+    }
+  }
+
+  const printedComments =
+    comments
+      .split(line)
+      .map(textLine => `${start} ${textLine}`)
+      .join(line) + line;
+
+  return (
+    head +
+    line +
+    (comments ? printedComments : '') +
+    (comments && keys.length ? start + line : '') +
+    printedObject +
+    tail
+  );
+}
+
+function printKeyValue(key, value) {
+  return `@${key} ${value}`.trim();
 }
 
-exports.extract = extract;
-exports.parse = parse;
+export default {extract, parse, parseWithComments, print};
```

Comment:
This was required for existing code that does `import docblock from 'jest-docblock'` to work without changes.

@SimenB should I leave this here or update the existing imports?

Reply:
Either way works for me. @cpojer thoughts?

Reply:
Fine to make a breaking change here, I'll publish 21.2 tomorrow or so, and for minor releases we can break the boundaries between Jest's modules.